### PR TITLE
`IResult` Implementation For `ContentResult`

### PR DESCRIPTION
**Pull request by ilkayilknur**
Contributes #32565

### CODE REVIEW

**Comment by ilkayilknur** [comment]
I see that you've been assigned for the other implementation PRs. You might want to review this PR as well. @halter73

File: src/Mvc/Mvc.Core/src/ContentResult.cs [hunk 1]
```diff
@@ -39,5 +44,56 @@ public override Task ExecuteResultAsync(ActionContext context)
...
+
+            var factory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = factory.CreateLogger<ContentResult>();
+
+            logger.ContentResultExecuting(resolvedContentType);
+
+            if (Content != null)
+            {
+                response.ContentLength = resolvedContentTypeEncoding.GetByteCount(Content);
+                var httpResponseStreamWriterFactory = httpContext.RequestServices.GetRequiredService<IHttpResponseStreamWriterFactory>();
```
Comment (halter73) [inline src/Mvc/Mvc.Core/src/ContentResult.cs:84]: Thanks for doing this! I cannot rely on a IHttpResponseStreamWriterFactory being registered for use with minimal Actions. Expecting the ILoggerFactory is fine. Let's use HttpResponseStreamWriter directly.

Comment (ilkayilknur) [reply src/Mvc/Mvc.Core/src/ContentResult.cs]: Thanks @halter73. I've updated the code.

**Review by halter73 — APPROVED** [review]
LGTM. Thanks!

File: src/Mvc/Mvc.Core/src/ContentResult.cs [hunk 2]
```diff
@@ -39,5 +45,55 @@ public override Task ExecuteResultAsync(ActionContext context)
...
+            var factory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = factory.CreateLogger<ContentResult>();
+
+            logger.ContentResultExecuting(resolvedContentType);
+
+            if (Content != null)
+            {
+                response.ContentLength = resolvedContentTypeEncoding.GetByteCount(Content);
+
+                await using (var textWriter = new HttpResponseStreamWriter(response.Body, resolvedContentTypeEncoding))
```
Comment (davidfowl) [inline src/Mvc/Mvc.Core/src/ContentResult.cs:86]: This looks inefficient. Can we just use Response.WriteAsync?

**Pull request merged**